cistem/protocols/protocol_ts_resample.py

```diff
@@ -0,0 +1,229 @@
+from pyworkflow import BETA
+from pyworkflow.protocol import PointerParam, IntParam
+from pyworkflow.utils import *
+from pyworkflow.object import Set
+from tomo.objects import SetOfTiltSeries, TiltSeries
+from pwem.emlib.image import ImageHandler
+
+OUTPUT_TS_NAME = 'resampledTiltSeries'
+OUTPUT_DIR = 'extra/'
+
+class ProtTsResample(EMProtocol, ProtTomoBase):
+    '''
+    Resample tilt series by Fourier cropping/padding using cisTEM. This is equivalent to binning/unbinning operations but free of aliasing artifacts.
+
+    More info:
+        https://cistem.org
+    '''
+
+    _label = 'resample tilt series'
+    _possibleOutputs = {OUTPUT_TS_NAME: SetOfTiltSeries}
+    _devStatus = BETA
+
+    tsList = []
+
+    def __init__(self, **args):
+        EMProtocol.__init__(self, **args)
+
+    # -------------------------- DEFINE param functions ----------------------
+    def _defineParams(self, form):
+        ''' Define the input parameters that will be used.
+        Params:
+            form: this is the form to be populated with sections and params.
+        '''
+        #     Example 'resample' input:
+        # 
+        # resample
+        #  
+        #  
+        #         **   Welcome to Resample   **
+        #  
+        #              Version : 1.00
+        #             Compiled : Dec  2 2017
+        #                 Mode : Interactive
+        #  
+        # Input image file name [ts2_L1G1-dose_filt.rec]   : ts2_L1G1-dose_filt.st
+        # Output image file name [ts2_L1G1_72_d3-bin8.rec] : ts2_L1G1-dose_filt-bin4.st
+        # Is the input a volume [YES]                        : NO
+        # New X-Size [464]                                   : 928
+        # New Y-Size [464]                                   : 928
+
+
+        # You need a params to belong to a section:
+        form.addSection(label=Message.LABEL_INPUT)
+        form.addParam('inputSetOfTiltSeries', PointerParam,
+                        pointerClass='SetOfTiltSeries',
+                        allowsNull=False,
+                        label='Input tilt series')
+
+        form.addParam('newXsize', IntParam,
+                        default=512,
+                        label='New X-Size',
+                        help='Volume will be rescaled to this size in X dimension (voxels)')
+        form.addParam('newYsize', IntParam,
+                        default=512,
+                        label='New Y-Size',
+                        help='Volume will be rescaled to this size in Y dimension (voxels)')
+
+    # -------------------------- INSERT steps functions -----------------------
+    def _insertAllSteps(self):
+
+        for ts in self.inputSetOfTiltSeries.get():
+
+            self._insertFunctionStep(self.runTsResample,
+                        ts.getObjId())
+
+            self._insertFunctionStep(self.createOutputStep,
+                        ts.getObjId())
+        
+        self._insertFunctionStep('closeStreamStep')
```

Comment:
this protocol will not work in streaming, because you are doing "for ts in self.inputSetOfTiltSeries.get():" above, i.e. you are getting the whole input set only once.